skimage/exposure/hdr.py

```diff
@@ -0,0 +1,318 @@
+import numpy as np
+
+
+def get_radiance(ims, exp, radiance_map):
+    """
+    Return the radiance for a series of images based upon a camera response 
+    function
+
+    Parameters
+    ----------
+    ims : list
+          list of images in the for of numpy arrays. Either mono or colour
+          (RGB, MxNx3)
+
+    exp : numpy 1D array
+          array of exposure times in seconds
+
+    radiacenMap : numpy array
+                  array mapping the counts to radiance
+
+    Returns
+    ----------
+    hdr : numpy array
+          resulting image with radiance values
+    """
+
+    den = np.ones(ims[0].shape)
+    num = np.zeros(ims[0].shape)
+    wf = np.vectorized(weight_func)
+    for idx, im in enumerate(ims):
+        gij = im.copy()
+        # For colour
+        if im.ndim == 3:
+            for ii in range(im.shape[2]):
+                gij[:, :, ii] = radiance_map[im[:, :, ii] + 1, ii]
+        else:
+            gij[:, :] = radiance_map[im[:, :, ii] + 1]
+        gij = gij - np.log(exp[idx])
+
+        wij = wf(zij)
+        num = num + wij * gij
+        den = den + wij
+
+    return num / den
+
+
+def make_hdr(ims, exp, radiance_map, depth=16):
+    """
+    Compute the HDR image from a series of images and a racianceMap
+    Based on:
+    Debevec, P. E., & Malik, J. (1997). SIGGRAPH 97 Conf. Proc., August, 3–8.
+    doi:10.1145/258734.258884
+
+    Parameters
+    ----------
+    ims : list
+          list of images in the for of numpy arrays. Either grayscale or colour
+          (RGB)
+
+    exp : numpy 1D array
+          array of exposure times in seconds
+
+    radiance_map : numpy array
+               array (idx) mapping counts to radiance value, if input is RGB 
+               this must be Nx3 
+
+    depth : int, optional
+            pixel depth, default=16
+
+    Returns
+    ----------
+    hdr : numpy array
+          The HDR image either grayscale or RGB depending on input in ln(E)
+    """
+    B = np.log(np.array(exp))
+
+    if ims[0].ndim == 3:
+        sx, sy, sc = np.shape(ims[0])
+        hdr = np.zeros([sx, sy, sc], dtype=np.float)
+        gray = False
+    else:
+        sx, sy = np.shape(ims[0])
+        hdr = np.zeros([sx, sy], dtype=np.float)
+        gray = True
+
+    ims = np.asarray(ims, dtype=np.int)
+
+    sx, sy, sz = np.shape(ims[0])
+
+    w = weight_func_arr(ims, depth)
+    for ii in range(sx):
+        for jj in range(sy):
+            if gray:
+                zij = ims[:, ii, jj]
+                g = radiance_map[zij]
+                hdr[ii, jj] = np.sum(
+                    w[:, ii, jj] * (g - B)) / np.sum(w[:, ii, jj])
+            else:
+                for cc in range(sc):
+                    zij = ims[:, ii, jj, cc]
+                    g = radiance_map[zij, cc]
+                    hdr[ii, jj, cc] = np.sum(
+                        w[:, ii, jj, cc] * (g - B)) / np.sum(w[:, ii, jj, cc])
+
```

Comment:
@paalge If a sum may resolve to zero on occasion, `max(result, 1)` or checking the result with an if statement and only normalizing if nonzero or greater than 1 are the usual approaches.